### PR TITLE
RGB Matrix Typing Heatmap

### PR DESCRIPTION
**Pull request by XScorpion2**
## Description

Requires https://github.com/qmk/qmk_firmware/pull/5783, just what the topic says, and just what @mechmerlin needs, the typing heatmap rgb effect. Also a few minor changes to digital rain. Will take another optimization pass at that effect soon.

Effect video: https://photos.app.goo.gl/GQvLdHA24eTR9hCk7

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Is it possible to lengthen the decay of the LED for this? 
Maybe use the "speed" option to increase how fast it decays?

**Comment by XScorpion2** [comment]
@drashna ya, integrating speed is something I've been trying to decide how to handle correctly. Right now it decays at 1 tick every 16ms so at max value it takes 4s to decay.

**Comment by drashna** [comment]
Well, this needs to be rebased. 

And as for the speed thing, maybe that can be added later.

**Comment by XScorpion2** [comment]
Ya, rebasing it now. Wife called me to dinner and is glaring at me tying this

**Comment by XScorpion2** [comment]
Rebased

**Comment by drashna** [comment]
Thanks!

File: quantum/rgb_matrix.c [hunk 1]
```diff
@@ -38,6 +38,7 @@
 #include "rgb_matrix_animations/rainbow_pinwheels_anim.h"
 #include "rgb_matrix_animations/rainbow_moving_chevron_anim.h"
 #include "rgb_matrix_animations/jellybean_raindrops_anim.h"
+#include "rgb_matrix_animations/typing_headmap_anim.h"
```
Comment (fauxpark) [inline quantum/rgb_matrix.c]: Typo in filename 😉

Comment (drashna) [reply quantum/rgb_matrix.c]: good catch!

Comment (drashna) [reply quantum/rgb_matrix.c]: Well, the file name is correct. As that's the files actual name.

Comment (fauxpark) [reply quantum/rgb_matrix.c]: Can't add reviews on filenames, I had to make do 😄

Comment (XScorpion2) [reply quantum/rgb_matrix.c]: will fix it in the cleanup pr I am working on now

**Comment by Lenbok** [comment]
I noticed when swapping from digital rain to heatmap that the keys initially lit up wrong, so some initialization is required like digital rain does if `params->init` is true.

**Comment by XScorpion2** [comment]
@Lenbok ehhhhh...does it really matter? =p
Will fix in next pr

**Comment by XScorpion2** [comment]
ok I lied, I fixed the effect init and file name in this pr

**Pull request merged**